### ISSUE TITLE
Regex character class does not allow unescaped hyphen at final position

### ISSUE THREAD

**Issue by woutersl** [cenotelie/hime#52]
**[Original report](https://bitbucket.org/cenotelie/hime/issue/52) by Jesper Lindholm (Bitbucket: [557058:60395acf-98ae-481a-ac02-8f761c9383fb](https://bitbucket.org/557058:60395acf-98ae-481a-ac02-8f761c9383fb), GitHub: [JesperTreetop](https://github.com/JesperTreetop)).**

----------------------------------------

The Hime grammar language reference says that regular expressions are supported and defers to [an IEEE standard](http://pubs.opengroup.org/onlinepubs/009695399/basedefs/xbd_chap09.html). In the 7th bullet point in section 9.3.5 RE Bracket Expression, which defined character classes, it says the following:

"The hyphen character shall be treated as itself if it occurs first (after an initial '^', if any) or last in the list, or as an ending range point in a range expression. As examples, the expressions "[-ac]" and "[ac-]" are equivalent and match any of the characters 'a', 'c', or '-'[..]"

In other words, including a hyphen without escaping it is permissible if done at the end. This is also congruent with every standard regex library I've used in C#, Ruby, Perl, PHP, etc.

But when I define a terminal like so:

`VARIABLE_NAME -> [A-Za-z0-9-]+;`

it does not match 'abc-xyz'. Whereas if I escape the hyphen:

`VARIABLE_NAME -> [A-Za-z0-9\-]+;`

it matches correctly. 

It is not a huge bug but as a silent discrepancy from both specs and common implementations it would be a good thing to not have to look out for.
